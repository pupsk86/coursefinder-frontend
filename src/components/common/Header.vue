<script>
import HeaderButtons from '@/components/common/HeaderButtons.vue'

export default {
    components: {
        HeaderButtons
    }
}
//TODO: add CompareButton component
//TODO: add FavouriteButton component
//TODO: add LoginButton component
</script>

<template>
    <header>
        <div class="header-container">
            <router-link to="/" class="logo"/>
            <header-buttons class="header-buttons"/>
        </div>
    </header>
</template>

<style scoped lang="less">
    header {
        height: 65px;
    }
    .header-container {
        position: fixed;
        width: 100%;
        left: 0px;
        right: 0px;
        top: 0px;
        border: 0;
        border-radius: 0;
        border-bottom: 1px solid rgba(0,0,0,.12);
        z-index: 3000;
        background-color: #ffffff;
    }
    .logo {
        margin: 5px 50px;
        width: 184px;
        float: left;
        height: 55px;
        background-image: url(../../assets/images/logo.png);
        background-size: 100%;
        background-repeat: no-repeat;
        background-position-y: -65px;
    }
    .header-buttons {
        margin: 25px 50px;
        float: right;
    }
</style>
