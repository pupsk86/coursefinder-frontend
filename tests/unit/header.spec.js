import { mount } from '@vue/test-utils'
import Header from '@/components/common/Header.vue'

describe('Header', () => {
  it('has a logo', () => {
    const wrapper = mount(Header)
    expect(wrapper.contains('.logo')).toBe(true)
  })
  it('renders correctly', () => {
    const wrapper = mount(Header)
    expect(wrapper.element).toMatchSnapshot()
  })
})
