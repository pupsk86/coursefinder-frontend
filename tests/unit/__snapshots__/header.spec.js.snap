// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    class="container"
  >
    <div
      class="logo"
    />
     
    <comparebutton />
     
    <favouritebutton />
     
    <loginbutton />
  </div>
</header>
`;
